scripts/ci-report/fetch_stats_from_ci.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Usage: ./fetch_stats_from_ci.sh
+#
+# This script is for fetching historic test stats from GitHub Actions CI.
```

Comment:
What is the next step for this script? I'm wondering if we can keep it as a Github Action. The result can be downloaded as workflow artifact.

Reply:
This script will hopefully not be needed much, it's main purpose is to populate the historical data since #6676. The main mechanism through which the gathering of stats will happen is via webhook.

It can still be useful in the future if someone wants to do some local analysis.